Spatial.gd

```diff
@@ -849,14 +855,19 @@ func new_trail_point_after(waypoint_trail: Waypoint.Trail, trail3d: Spatial, tra
 
 func refresh_trail3d_points(trail3d: Spatial):
 	trail3d.refresh_mesh()
+	set_unsaved_changes(true)
 
 func refresh_trail2d_points(trail2d: Line2D):
 	trail2d.refresh_points()
+	set_unsaved_changes(true)
```

Comment:
This seems like the wrong place to trigger the fact that the proto data has been changed.